### ISSUE TITLE
Carousel caption misbehavior when there's insufficient room for the text

### ISSUE THREAD

**Issue by moinakbarali** [twbs/bootstrap#18140]
Docs says that carousel will automatically align captions and format it. but if we add large content into caption it does not align properly.
I thought it will get dynamic position like in Popovers and tooltip that have an option to align them automatically. 
I look through carousel code and does not find anything that automatically align content within caption, so this section should not say that.
http://getbootstrap.com/javascript/#optional-captions
![untitled](https://cloud.githubusercontent.com/assets/2787717/10907921/c318819e-824f-11e5-97f7-497d67c43684.png)

**Comment by cvrebert** [comment]
Could you post a JS Bin or JSFiddle that demonstrates the misalignment?

**Comment by moinakbarali** [comment]
https://jsbin.com/cuhoyaduli/edit?html,output
check this on 320px width.

**Comment by twbs-lmvtfy** [comment]
Hi @moinakbarali!

You appear to have posted a live example (https://jsbin.com/cuhoyaduli/edit), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* [W002](https://github.com/twbs/bootlint/wiki/W002): `<head>` is missing X-UA-Compatible `<meta>` tag that disables old IE compatibility modes
* [W003](https://github.com/twbs/bootlint/wiki/W003): `<head>` is missing viewport `<meta>` tag that enables responsiveness

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by cvrebert** [comment]
So, the actual bug seems to be that slide images aren't centered when they're narrower than the carousel itself.

**Comment by bassjobsen** [comment]
As far as i do understand  @moinakbarali; the only problem is that the text is too long to fit the carousel?
Possible related: https://github.com/twbs/bootstrap/issues/18171

**Comment by moinakbarali** [comment]
Yes @bassjobsen is correct. But as the carousel doc say that it will align content automatically it should handle long text someway or it should not say that it will align content automatically.

@cvrebert that is also an issue, I fixed them using `margin: 0 auto` or by adding `centered-block` class to image.

**Comment by cvrebert** [comment]
Closing this as a duplicate of #18171 then.
I'll open a new issue for the image centering problem.
